src/ResourceGraph/ResourceGraph/Utilities/ResultExtensions.cs

```diff
@@ -49,41 +42,63 @@ public static class TableExtensions
             typeof(PSCustomObject).FullName + "#QueryResponse";
 
         /// <summary>
-        /// Converts table to the PS Objects.
+        /// Converts object to the PS Objects.
         /// </summary>
-        /// <param name="table">The table.</param>
+        /// <param name="data">The data.</param>
         /// <returns></returns>
-        public static IEnumerable<PSObject> ToPsObjects(this Table table)
+        public static IList<PSObject> ToPsObjects(this object data)
         {
-            var idColumnIndex = IndexOf(table.Columns, column =>
-                string.Equals(column.Name, IdColumnName, StringComparison.OrdinalIgnoreCase) &&
-                column.Type == IdColumnType);
+            try
+            {
+                if (data is JArray array)
+                {
+                    return ToPsObjects(array.ToList());
+                }
 
-            foreach (var row in table.Rows)
+                throw new ArgumentOutOfRangeException("Result format is not supported");
```

Comment:
This exception will be double thrown. Can we optimize?

Reply:
Done